squash error handling into debugging-queries (#92)
chalk-ai/docs: debugging-queries.mdx

You can also view these logs using `resolver_replay`, where they will be emitted to your terminal:

![Logs output example](/img/debugging/logs.png)

## Query Error Categories

In using the debugging techniques above, you may encounter a few different categories of errors
that have common root causes:

### Request error

> Request errors are raised before execution of your resolver code.

Reviewer: ```suggestion
Request errors are raised before your resolver code executes.
```

> They may occur due to invalid feature names in the

Reviewer: ```suggestion
They are often caused by invalid feature names in the
```

> input or a request that cannot be satisfied by the resolvers you

Reviewer: ```suggestion
input or by requests that cannot be satisfied by the resolvers you
```

have defined.

### Field error

Field errors are raised while running a feature resolver

> for a particular field.

Reviewer: What is meant by for a particular field, here? Is this like what we have sort of moved to calling feature—like a single feature in a feature set? Maybe something like:

"Feature errors are raised when a specific resolver fails"?

For this type of error, you'll find a `feature` and `resolver`
attribute in the [error type](#chalkerror).

> When a feature resolver crashes, you will receive `null` value

Reviewer: ```suggestion
When a resolver crashes, you will receive `null` value
```
Maybe generally just resolver? But IDK haha

in the response.
To differentiate from a resolver returning a `null` value and
a failure in the resolver, you need to check the error schema.

### Network error

Network errors are thrown outside your resolvers.

> For example, your request was unauthenticated,

Reviewer: ```suggestion
They can be caused by unauthenticated requests,
```

> connection failed, or an error occurred within Chalk.

Reviewer: ```suggestion
connection failures, or errors occurring within Chalk.
```
Should we remove errors occurring within Chalk haha or find some way to say it more delicately haha :)

Author: i will ~ refine ~
